### PR TITLE
CFEP-18: Remove static libraries from main build

### LINKED ISSUE #260: Split out static library
The libpython static library is quite large

```
$ du -h ./lib/libpython3.7m.a
 16M    ./lib/libpython3.7m.a
```

and won't be used by most users. It would be useful to split this out into a separate package if possible, to reduce installation size (important for docker images).

### PR DESCRIPTION
**Pull request by xhochy**
Fixes #260 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xhochy** [comment]
Also seeing the Linux error in https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=189743&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=791 in #366 which is not using outputs, so this might be unrelated.

**Comment by beckermr** [comment]
These lines are very odd. The old sysroot paths should not be in this build.

```
checking build system type... x86_64-conda_cos6-linux-gnu
checking host system type... x86_64-conda_cos6-linux-gnu
```

**Comment by beckermr** [comment]
@xhochy This failure is due to a bug in the compiler activation for the new sysroot. It should be fixed within a few hours or so.

**Comment by mingwandroid** [comment]
Great stuff!

**Comment by xhochy** [comment]
The Linux issue is the same as in #368 , continuing with Windows here then.

**Comment by xhochy** [comment]
Verified that the Windows build currently works with the same script, just without using `outputs:`, so this is probably an `outputs` problem / different behaviour.

**Comment by xhochy** [comment]
@conda-forge/python Any idea why this fails on Windows?

**Comment by xhochy** [comment]
Thanks @mbargull!

**Comment by xhochy** [comment]
@conda-forge/python This is ready for review/merge.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -116,83 +116,119 @@ build:
...
+        -   pushd prefix-replacement  # [unix]
+        -     bash build-and-test.sh  # [unix]
+        -   popd  # [unix]
+        -   pushd cmake
+        -     cmake -GNinja -DPY_VER={{ version }}
+              # --trace --debug-output --debug-trycompile .
+        -   popd
+        - popd
+        - test ! -f default.profraw   # [osx]
+  - name: libpython-static
```
Comment (scopatz) [inline recipe/meta.yaml:213]: Shouldn't this just be `python-static` to follow the convention?

Comment (xhochy) [reply recipe/meta.yaml:213]: That would be a bit confusing in this case as the main Python is also statically linking libpython. As it's only the lib what is included in the -static package, I would rather reflect that also in the name.

Comment (scopatz) [reply recipe/meta.yaml:213]: It still seems a bit odd to me.  I am not blocking based on the name though.

Comment (xhochy) [reply recipe/meta.yaml:213]: Any other opinions? I would happily sacrifice the name for a merge but I still think `libpython-static` is more intuitive regarding the contents of the new output.

File: recipe/build_static.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+if [[ ${DEBUG_PY} == yes ]]; then
+  DBG=d
+else
+  DBG=
+fi
+VER=${PKG_VERSION%.*}
+VERABI=${VER}${DBG}
+
+
+cp -pf ${_buildd_static}/libpython${VERABI}.a ${PREFIX}/lib/libpython3.8.a
```
Comment (mariusvniekerk) [inline recipe/build_static.sh:15]: The 3.8 here probably shouldn't be explicit

Comment (xhochy) [reply recipe/build_static.sh]: Fixed and CI passed successfully.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -116,83 +116,119 @@ build:
...
+        - posix  # [win]
+      host:
+        - {{ pin_subpackage('python', exact=True) }}
+      run:
+        - {{ pin_subpackage('python', exact=True) }}
+    test:
+      commands:
+        - test -f ${PREFIX}/lib/libpython3.8.a  # [unix]
+        - test -f ${PREFIX}/lib/libpython3.8.nolto.a  # [unix]
+        - test -f ${PREFIX}/lib/python3.8/config-3.8-darwin/libpython3.8.a  # [osx]
```
Comment (scopatz) [inline recipe/meta.yaml:231]: explicit 3.8 mentions here

File: recipe/meta.yaml (hunk 1 above)
Comment (scopatz) [reply recipe/meta.yaml:213]: I am also happy to be outvoted here too.

**Comment by xhochy** [comment]
@conda-forge/python ping again here, merging this would be a massive improvement as it should reduce the on-disk size of a basic Python installation by 25%.

**Pull request merged**